### PR TITLE
#28 Add GSI/LSI creation when creating a table

### PR DESCRIPTION
**Pull request by ryankazokas**
- Adds the interface to be able to create secondary indexes when creating a table
- Changes form processing to use JSON data and error handling more async.

### CODE REVIEW

File: views/create-table.ejs [hunk 1]
```diff
@@ -1,85 +1,251 @@
...
-                        <input type="text" id="TableName" name="TableName" class="form-control" autofocus/>
-                    </div>
+    <header>
+        <ol class="breadcrumb">
+            <a class="breadcrumb-item" href="/">Tables</a>
+            <a class="breadcrumb-item active">Create Table</a>
+        </ol>
+    </header>
+    <main>
+        <div class="list-items"></div>
```
Comment (rchl) [inline views/create-table.ejs]: Unused, I believe.

File: views/create-table.ejs [hunk 2]
```diff
@@ -1,85 +1,251 @@
...
+                        <div class="form-group">
+                            <label for="IndexName">Index Name</label>
+                            <input type="text" name="secondary-index-${index}-IndexName" class="form-control" autofocus />
+                        </div>
+                    </div>
+                    <div class="col-md-6">
+                        <div class="form-group">
+                            <label for="IndexType">Index Type</label>
+                            <select name="secondary-index-${index}-IndexType" class="form-control">
+                                <option></option>
```
[the remark this answers is not in the document]
Comment (ryankazokas) [reply views/create-table.ejs]: I agree. I was just keeping uniform with the Index Types. I'll update to make gsi default

File: lib/backend.js [hunk 3]
```diff
@@ -193,19 +193,83 @@ exports.createServer = (dynamodb, docClient) => {
...
+              KeyType: 'RANGE'
+            })
+          }
+          const index = {
+            IndexName: secondaryIndex.IndexName,
+            KeySchema: secondaryIndexKeySchema,
+            Projection: {
+              ProjectionType: 'ALL'
+            },
+            ProvisionedThroughput: {
```
[the remark this answers is not in the document]
Comment (ryankazokas) [reply lib/backend.js]: You are right. It uses the base table capacity. I've disabled the read capacity/write capacity fields when LSI is selected and added some frontend and back end checks here.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dynamodb-admin",
-  "version": "3.1.1",
+  "version": "3.2.0",
```
[the remark this answers is not in the document]
Comment (ryankazokas) [reply package.json]: My bad, i've worked on projects where it's part of the contibutors responsibility. I've reverted this back to 3.1.1

**Comment by ryankazokas** [comment]
@rchl i ran with the --fix option before submitting before my last commit.

**Comment by rchl** [comment]
Thank you again. :)

**Pull request merged**